src/main/java/org/terasology/projsndwv/TempBeeRegistry.java

```diff
@@ -37,32 +42,62 @@
     private static HashMap<Integer, Integer> lifespans = new HashMap<>();
     private static HashMap<Integer, Long> tickspeeds = new HashMap<>();
     private static HashMap<Integer, Production> productions = new HashMap<>();
+    private static HashMap<Integer, HashMap<Integer, String>> genotypeNames = new HashMap<>();
 
     private static MersenneRandom random;
 
     private static Logger logger = LoggerFactory.getLogger(TempBeeRegistry.class);
 
-    public static TextureRegionAsset<?> getTextureRegionAssetForSpeciesAndType(int species, int type) {
+    public static EntityRef modifyItemForSpeciesAndType(EntityRef entity) {
+        if (!entity.hasComponent(BeeComponent.class)) {
+            return entity;
+        }
+
+        int species = entity.getComponent(GeneticsComponent.class).activeGenes.get(ApiarySystem.LOCUS_SPECIES);
+        BeeComponent.BeeType type = entity.getComponent(BeeComponent.class).type;
+
+        String typeName = "NULL";
+        switch(type) {
+            case DRONE:
+                typeName = "Drone";
+                break;
+            case PRINCESS:
+                typeName = "Princess";
+                break;
+            case QUEEN:
+                typeName = "Queen";
+                break;
+        }
+
         AssetManager assetManager = CoreRegistry.get(AssetManager.class);
         if (assetManager == null) {
             logger.error("No AssetManager in registry");
             return null;
         }
 
         // TODO: This call can probably be checked, but it'll take a serious refactor.
-        Optional<TextureRegionAsset> textureRegionAsset = assetManager.getAsset("Apiculture:bee_" + new String[]{"a", "b", "c"}[species] + "_" + new String[]{"drone", "princess", "queen"}[type], TextureRegionAsset.class);
+        Optional<TextureRegionAsset> textureRegionAsset = assetManager.getAsset("Apiculture:bee_" + new String[]{"a", "b", "c"}[species] + "_" + typeName.toLowerCase(), TextureRegionAsset.class);
         if (!textureRegionAsset.isPresent()) {
-            logger.error("Texture not found for species " + species + " and type " + type);
+            logger.error("Texture not found for species " + species + " and type '" + typeName + "'");
             return null;
         }
 
-        return textureRegionAsset.get();
-    }
+        ItemComponent itemComponent = entity.getComponent(ItemComponent.class);
```

Comment:
Maybe another place to harden here a bit, in case there's no `ItemComponent` for some reason

Another further down with `DisplayNameComponent` but I'll stop noting these for now :-)